smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/CodegenVisitor.java

```diff
@@ -277,10 +278,30 @@ public Void serviceShape(ServiceShape shape) {
         // Generate each operation for the service.
         TopDownIndex topDownIndex = model.getKnowledge(TopDownIndex.class);
         Set<OperationShape> containedOperations = new TreeSet<>(topDownIndex.getContainedOperations(service));
+        Set<String> paginatorLocations = new TreeSet<>();
+        String paginationInterfaceLocation = "pagination/Interfaces.ts";
+
         for (OperationShape operation : containedOperations) {
             writers.useShapeWriter(operation, commandWriter -> new CommandGenerator(
                     settings, model, operation, symbolProvider, commandWriter,
                     runtimePlugins, protocolGenerator, applicationProtocol).run());
+            if(operation.hasTrait(PaginatedTrait.ID)){
+                String outputFilename = "pagination/"+ operation.getId().getName() +"Paginator.ts";
+                paginatorLocations.add(outputFilename);
+                writers.useFileWriter(outputFilename, paginationWriter ->
+                        new PaginationGenerator(settings, model, operation, symbolProvider, paginationWriter,
+                                runtimePlugins, protocolGenerator, applicationProtocol, nonModularName, paginationInterfaceLocation).run());
+            }
+        }
+
+        // TODO add pagination files to export in index.ts
+        for (String file: paginatorLocations) {
+            // write to the index.ts an export
```

Comment:
Is there a clean way of including all the paginated files in the index.ts?

Reply:
The `index.ts` file is written in the [`IndexGenerator`](https://github.com/awslabs/smithy-typescript/blob/master/smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/IndexGenerator.java), so you could abstract the pagination file's name generation and update that to include the right files. Doing so would allow you to move the `paginatorLocations` List to a boolean as well, since this loop wouldn't be needed here.

Reply:
done